crates/binjs_es6/src/scopes.rs

```diff
@@ -20,13 +23,22 @@ struct VarAndLexNames {
     const_lexical_names: HashSet<String>,
 }
 
+enum ParamKind {
+    FunctionName { name: String },
+    Positional { index: u32, name: String },
+    Destructuring { name: String },
+    Rest { name: String },
+}
+
 #[derive(Default)]
 pub struct AnnotationVisitor {
     // The following are stacks.
     var_names_stack: Vec<HashSet<String>>,
     non_const_lexical_names_stack: Vec<HashSet<String>>,
     const_lexical_names_stack: Vec<HashSet<String>>,
-    param_names_stack: Vec<HashSet<String>>,
+    params_stack: Vec<Vec<ParamKind>>,
+    param_indices_stack: Vec<u32>,
```

Comment:
Yes, I'll add document.
It's for tracking the current parameter index for each (nested) function.
(since there's no other simple way to check the current parameter index from the visitor)